pkg/pusher/pusher_test.go

```diff
@@ -0,0 +1,174 @@
+	_, storer := createPusher(t, triggerPeer, pushSyncService, mock.WithClosestPeer(closestPeer))
+
+	_, err := storer.Put(context.Background(), storage.ModePutUpload, chunk)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Give some time for chunk to be pushed and receipt to be received
+	time.Sleep(10 * time.Millisecond)
+
+	err = checkIfModeSet(t, chunk.Address(), storage.ModeSetSyncPush, storer)
+	if err == nil {
+		t.Fatalf("chunk not syned error expected")
+	}
+}
+
+func TestSendChunkAndTimeoutinReceivingReceipt(t *testing.T) {
+	chunk := createChunk(t)
+
+	// create a trigger  and a closestpeer
+	triggerPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
+	closestPeer := swarm.MustParseHexAddress("6000000000000000000000000000000000000000000000000000000000000000")
+
+	pushSyncService := pushsyncmock.New(func(ctx context.Context, address swarm.Address, chunk swarm.Chunk) (*pb.Receipt, error) {
+		// Set 10 times more than the time we wait for the test to complete so that
+		// the response never reaches our testcase
+		time.Sleep(100 * time.Millisecond)
+		return nil, nil
+	})
+
+	_, storer := createPusher(t, triggerPeer, pushSyncService, mock.WithClosestPeer(closestPeer))
+
+	_, err := storer.Put(context.Background(), storage.ModePutUpload, chunk)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Give some time for chunk to be pushed and receipt to be received
+	time.Sleep(10 * time.Millisecond)
+
+	err = checkIfModeSet(t, chunk.Address(), storage.ModeSetSyncPush, storer)
+	if err == nil {
+		t.Fatalf("chunk not syned error expected")
+	}
+}
+
+func createChunk(t *testing.T) swarm.Chunk {
+	t.Helper()
+
+	// chunk data to upload
+	chunkAddress := swarm.MustParseHexAddress("7000000000000000000000000000000000000000000000000000000000000000")
+	chunkData := []byte("1234")
+	return swarm.NewChunk(chunkAddress, chunkData)
+}
+
+func createPusher(t *testing.T, addr swarm.Address, pushSyncService pushsync.PushSyncer, mockOpts ...mock.Option) (*pusher.Service, *Store) {
+	t.Helper()
+	logger := logging.New(ioutil.Discard, 0)
+	storer, err := localstore.New("", addr.Bytes(), nil, logger)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pusherStorer := &Store{
+		Storer:    storer,
+		ModeSet:   make(map[string]storage.ModeSet),
+		modeSetMu: &sync.Mutex{},
+	}
+	peerSuggester := mock.NewTopologyDriver(mockOpts...)
+	pusherService := pusher.New(pusher.Options{Storer: pusherStorer, PushSyncer: pushSyncService, PeerSuggester: peerSuggester, Logger: logger})
+	return pusherService, pusherStorer
+}
+
+func checkIfModeSet(t *testing.T, addr swarm.Address, mode storage.ModeSet, storer *Store) error {
+	t.Helper()
+
+	var found bool
+	storer.modeSetMu.Lock()
+	defer storer.modeSetMu.Unlock()
+
+	for k, v := range storer.ModeSet {
+		if addr.String() == k {
+			found = true
+			if v != mode {
+				return errors.New("chunk mode is not properly set as synced")
+			}
+		}
+	}
+	if !found {
+		return errors.New("Chunk not synced")
```

Comment:
you can fail directly here with `t.Fatal`, otherwise remove `t`

Reply:
removed t  as said in the earlier comment.